sql: TestCancelQueriesRace is flaky

First observed this failure in a PR that changed unrelated code [1]. I was also able to reproduce the failure by stressing the test with the command below:

```console
./dev test ./pkg/sql -f TestCancelQueriesRace --stress --ignore-cache --cpus 16 --timeout 10m
--- FAIL: TestCancelQueriesRace (1.55s)
    test_log_scope.go:167: test logs captured to: /tmp/cockroach/_tmp/b287de7194e17a0386ee1775f0d198ba/logTestCancelQueriesRace219461028
    test_log_scope.go:81: use -show-logs to present logs inline
    show_test.go:1351: 
                Error Trace:    github.com/cockroachdb/cockroach/pkg/sql_test/pkg/sql/show_test.go:1351
                Error:          Received unexpected error:
                                pq: could not cancel query 176d3f136c1d12710000000000000001: query ID 176d3f136c1d12710000000000000001 not found
                Test:           TestCancelQueriesRace
    panic.go:522: -- test log scope end --
FAIL
I230629 21:39:48.032652 1 (gostd) testmain.go:1030  [-] 1  Test //pkg/sql:sql_test exited with error code 1


ERROR: exit status 1

421 runs completed, 1 failures, over 1m3s
```

[1] <https://teamcity.cockroachdb.com/viewLog.html?buildId=10734269&buildTypeId=Cockroach_BazelEssentialCi>

Jira issue: CRDB-29235

--- a/pkg/sql/show_test.go
+++ b/pkg/sql/show_test.go
@@ -1341,14 +1341,18 @@ func TestCancelQueriesRace(t *testing.T) {
 		_, _ = sqlDB.ExecContext(ctx, `SELECT pg_sleep(10)`)
 		close(waiter)
 	}()
-	_, err := sqlDB.ExecContext(ctx, `CANCEL QUERIES (
+	_, err1 := sqlDB.ExecContext(ctx, `CANCEL QUERIES (
 		SELECT query_id FROM [SHOW QUERIES] WHERE query LIKE 'SELECT pg_sleep%'
 	)`)
-	require.NoError(t, err)
-	_, err = sqlDB.ExecContext(ctx, `CANCEL QUERIES (
+
+	_, err2 := sqlDB.ExecContext(ctx, `CANCEL QUERIES (
 		SELECT query_id FROM [SHOW QUERIES] WHERE query LIKE 'SELECT pg_sleep%'
 	)`)
-	require.NoError(t, err)
+	// At least one query cancellation is expected to succeed.
+	require.Truef(
+		t,
+		err1 == nil || err2 == nil,
+		"Both query cancellations failed with errors: %v and %v", err1, err2)
 
 	cancel()
 	<-waiter

